dbdimp.c

```diff
@@ -1643,9 +1643,32 @@ static bool mariadb_dr_connect(
   {
     if (imp_drh->instances == 0 && !imp_drh->non_embedded_started)
     {
+      int init_failed;
+
+      /*
+       * MariaDB Connector/C 3.0.0 in function mysql_server_init() for non-Windows systems started
+       * setting SIGPIPE handler to SIG_IGN. This breaks Perl applications which installed its own
+       * SIGPIPE signal handler. As a workaround use Perl rsignal_save() function to save existing
```

Comment:
This also breaks graceful termination of programs using the default signal behavior.